src/shogun/kernel/GaussianARDKernel.cpp

```diff
@@ -124,7 +124,7 @@ float64_t CGaussianARDKernel::get_parameter_gradient_helper(
 
 	if (!strcmp(param->m_name, "weights"))
 	{
-		linalg::add(avec, bvec, bvec, 1.0, -1.0);
+		bvec=linalg::add(avec, bvec, 1.0, -1.0);
```

Comment:
In place would be good in general, but keep in mind this is just a vector operation (and the vector is not large)
In-place is more crucial when we create copies of the kernel matrices (which are the main memory bottleneck of GPs)